### ISSUE TITLE
fast-scnn训练后CPU推理速度一张图需要2.5秒左右，速度慢很多  有办法加速吗？请问有什么办法提高精度？

### ISSUE THREAD

**Issue by jaffe-fly** [PaddlePaddle/PaddleSeg#201]
`DEPLOY:
        USE_GPU : 0
        USE_PR : 1
        MODEL_PATH : "E:/fire_detection/fire_code/freeze_model"
        MODEL_FILENAME : "__model__"
        PARAMS_FILENAME : "__params__"
        EVAL_CROP_SIZE : (1920, 1080)
        MEAN : [0.5, 0.5, 0.5]
        STD : [0.5, 0.5, 0.5]
        IMAGE_TYPE : "rgb"
        NUM_CLASSES : 2
        CHANNELS : 3
        PRE_PROCESSOR : "SegPreProcessor"
        PREDICTOR_MODE : "ANALYSIS"
        BATCH_SIZE : 1
    `

另外训练集只有200张图片，训练出来效果很差，训练了100个epoch，请问有什么办法提高精度

**Comment by sjtubinlong** [comment]
@jaffe-fly 你好，关于推理速度，试试增大batch。另外前几个batch会相对慢一些，后续的速度会快一些。

**Comment by jaffe-fly** [comment]
我是直接处理视频流的 读一帧处理一帧这种 所以没有batch

**Comment by LutaoChu** [comment]
关于精度，因为训练集图片数量较少，可以尝试使用数据增强，如stepscaling、rangescaling、mirror等。loss推荐使用dice loss+bce loss  https://github.com/PaddlePaddle/PaddleSeg/blob/release/v0.4.0/docs/loss_select.md

**Comment by jaffe-fly** [comment]
这些方法都用了

**Comment by LutaoChu** [comment]
可以提供一下训练yaml吗

**Comment by jaffe-fly** [comment]
`TRAIN_CROP_SIZE: (512, 512) # (width, height), for unpadding rangescaling and stepscaling
#EVAL_CROP_SIZE: (1997, 1024) # (width, height), for unpadding rangescaling and stepscaling
EVAL_CROP_SIZE: (1920, 1080) # (width, height), for unpadding rangescaling and stepscaling
AUG:
    AUG_METHOD: "unpadding" # choice unpadding rangescaling and stepscaling
    FIX_RESIZE_SIZE: (512, 512) # (width, height), for unpadding

    #INF_RESIZE_VALUE: 500  # for rangescaling
    #MAX_RESIZE_VALUE: 600  # for rangescaling
    #MIN_RESIZE_VALUE: 400  # for rangescaling

    #MAX_SCALE_FACTOR: 1.25  # for stepscaling
    #MIN_SCALE_FACTOR: 0.75  # for stepscaling
    #SCALE_STEP_SIZE: 0.25  # for stepscaling
 # 图像镜像左右翻转
    MIRROR: True
    # 图像上下翻转开关，True/False
    FLIP: True
    # 图像启动上下翻转的概率，0-1
    FLIP_RATIO: 0.5

    RICH_CROP:
        # RichCrop数据增广开关，用于提升模型鲁棒性
        ENABLE: True
        # 图像旋转最大角度，0-90
        MAX_ROTATION: 15
        # 裁取图像与原始图像面积比，0-1
        MIN_AREA_RATIO: 0.5
        # 裁取图像宽高比范围，非负
        ASPECT_RATIO: 0.33
        # 亮度调节范围，0-1
        BRIGHTNESS_JITTER_RATIO: 0.5
        # 饱和度调节范围，0-1
        SATURATION_JITTER_RATIO: 0.5
        # 对比度调节范围，0-1
        CONTRAST_JITTER_RATIO: 0.5
        # 图像模糊开关，True/False
        BLUR: True
        # 图像启动模糊百分比，0-1
        BLUR_RATIO: 0.1
BATCH_SIZE: 32
DATASET:
    DATA_DIR: r"E:\fire_detection\fire_code\home\aistudio\work\dataset"
    IMAGE_TYPE: "rgb"  # choice rgb or rgba
    NUM_CLASSES: 2
    TEST_FILE_LIST: r"E:\fire_detection\fire_code\home\aistudio\work\dataset\test.txt"
    TRAIN_FILE_LIST: r"E:\fire_detection\fire_code\home\aistudio\work\dataset\train.txt"
    VAL_FILE_LIST: r"E:\fire_detection\fire_code\home\aistudio\work\dataset\val.txt"
    VIS_FILE_LIST: r"E:\fire_detection\fire_code\home\aistudio\work\dataset\test.txt"
    IGNORE_INDEX: 255
    SEPARATOR: " "
FREEZE:
    MODEL_FILENAME: "__model__"
    PARAMS_FILENAME: "__params__"
MODEL:
    MODEL_NAME: "fast_scnn"
    DEFAULT_NORM_TYPE: "bn"

TRAIN:
    PRETRAINED_MODEL_DIR: "./pretrained_model/fast_scnn_cityscapes/"
    MODEL_SAVE_DIR: "./saved_model/fast_scnn_pet/"
    SNAPSHOT_EPOCH: 10
TEST:
    TEST_MODEL: "./saved_model/fast_scnn_pet/final"
SOLVER:
    NUM_EPOCHS: 100
    LR: 0.005
    LR_POLICY: "poly"
    OPTIMIZER: "sgd"
    LOSS: ["bce_loss", "dice_loss"]
`

**Comment by LutaoChu** [comment]
有几个建议可以参考一下

- TRAIN_CROP_SIZE太小，调大一些

- 关闭FLIP，这个场景不太适用

- 设置AUG_METHOD 为step_scaling

- 关闭RICH_CROP.ENABLE

- 尝试其他的learning rate

**Comment by jaffe-fly** [comment]
![image](https://user-images.githubusercontent.com/49515380/77217273-2f1fd500-6b5c-11ea-9426-e0c9d3fe713e.png)
图像分割的数据集标签，作为二分类来说 背景就是负样本，是否有必要出现这种没有目标的图（全黑的图）?

**Comment by LutaoChu** [comment]
你可以观察一下评估的category iou，如果背景类的识别准确率不够高，可以加上这些图片，加强模型对背景的识别能力。如果已经很高了，但前景的识别效果不理想，可以去掉这些负样本。

**Comment by jaffe-fly** [comment]
[EVAL]Category Acc: [0.9930 0.5560]
[EVAL]Kappa:0.6324
这个是训练的最好结果了  但实际视频变现还很很差，在没有目标的帧上，也出现误识别，还有其他方式可以继续提高吗？

**Comment by jaffe-fly** [comment]
或者怎么可以抑制背景的识别精度？

**Comment by LutaoChu** [comment]
看一下category iou的指标吧
最好也能提供一些识别效果图和badcase

**Comment by jaffe-fly** [comment]
[EVAL]#image=61 acc=0.9755 IoU=0.7253
[EVAL]Category IoU: [0.9750 0.4757]
[EVAL]类别Acc：[0.9930 0.5560]
[EVAL] Kappa：0.6324

**Comment by jaffe-fly** [comment]
目标是识别火  已经附件了识别结果 请查看

**Comment by LutaoChu** [comment]
> 目标是识别火 已经附件了识别结果 请查看

没有看到附件，是撤回了吗？

另外方便的话也提供下目前的训练配置吧

**Comment by jaffe-fly** [comment]
由于之前网络的精度问题，我放弃了换成HRNET_W64，网络推理时间应该是没错的，要想提高就要配合slim实现了，所以不纠结这个时间了
